controllers/clustertopology_controller_compute.go

```diff
@@ -140,9 +141,129 @@ func (r *ClusterTopologyReconciler) getClass(ctx context.Context, cluster *clust
 	return class, nil
 }
 
-// getTemplate gets the object referenced in ref.
+// getCurrentState gets information about the current state of a cluster by inspecting the state of the cluster infrastructure,
+// the control plane, and the machine deployments associated with the cluster.
+func (r *ClusterTopologyReconciler) getCurrentState(ctx context.Context, cluster *clusterv1.Cluster, class clusterv1.ClusterClass) (*clusterTopologyState, error) {
+	var err error
+	var infra *unstructured.Unstructured
+	var cp controlPlaneTopologyState
+	var m []machineDeploymentTopologyState
+	// Reference to the cluster infrastructure can be nil and is expected to be on the first reconcile loop
+	// In this case the method should still be allowed to continue
+	if cluster.Spec.InfrastructureRef != nil {
+		infra, err = r.getCurrentInfrastructureClusterState(ctx, cluster)
+		if err != nil {
+			return nil, errors.Wrap(err, "current cluster state not found")
```

Comment:
```suggestion
			return nil, errors.Wrap(err, "failed to retrieve Cluster state")
```
What do you think about this error msg? I think it would be more consistent with the others we already have

(would also affect other errors below)

Reply:
👍